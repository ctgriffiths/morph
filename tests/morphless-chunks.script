#!/bin/sh
# Try to build a morphless chunk.
# Copyright (C) 2012  Codethink Limited
# 
# This program is free software; you can redistribute it and/or modify
# it under the terms of the GNU General Public License as published by
# the Free Software Foundation; version 2 of the License.
# 
# This program is distributed in the hope that it will be useful,
# but WITHOUT ANY WARRANTY; without even the implied warranty of
# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
# GNU General Public License for more details.
# 
# You should have received a copy of the GNU General Public License along
# with this program; if not, write to the Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor, Boston, MA 02110-1301 USA.


set -eu


# Create an auto-detectable chunk.

mkdir "$DATADIR/morphless"
cd  "$DATADIR/morphless"

touch configure
chmod +x configure

cat << EOF > Makefile
all install:
EOF

git init --quiet
git add *
git commit --quiet -m initial


# Create a stratum that refers to the morphless chunk.

mkdir "$DATADIR/stratum"
cd  "$DATADIR/stratum"
cat << EOF > alfred.morph
{
    "name": "alfred",
    "kind": "stratum",
    "sources": [
        {
            "name": "morphless",
            "ref": "master"
        }
    ]
}
EOF
git init --quiet
git add alfred.morph
git commit --quiet -m initial

"$SRCDIR/scripts/test-morph" build stratum master alfred.morph

for chunk in "$DATADIR/cache/"*.chunk.*
do
    echo "$chunk:" | sed 's/[^.]*//'
    tar -tf "$chunk" | LC_ALL=C sort | sed '/^\.\/./s:^\./::'
    echo
done
