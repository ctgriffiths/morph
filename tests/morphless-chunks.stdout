.chunk.morphless:
./
baserock/
baserock/morphless.meta

